### PR TITLE
Add ability to show next increment version

### LINKED ISSUE #86: It would be useful to have next_version accessible as a public method
Here is our CI [script](https://github.com/rubykube/peatio/blob/571ac4bb685bff3eccdbfe82f66e9650a030f2d7/lib/tasks/release.rake#L30) and we need to use send because there is no ability to get next version without tagging

My suggestion is to move next_version method to public

I will submit PR for this change

### PR DESCRIPTION
**Pull request by ysv**
closes #86

### CODE REVIEW

**Comment by ysv** [comment]
@gregorym @grosser Hello!

It's WIP but still could you have a look please

File: lib/bump.rb [hunk 1]
```diff
@@ -220,31 +250,6 @@ def find_version_file(pattern)
         end
       end
 
-      def next_version(current, part)
```
Comment (gregorym) [inline lib/bump.rb]: What actually changed here? Indentation?

File: bump.gemspec [hunk 2]
```diff
@@ -14,4 +14,5 @@ Gem::Specification.new "bump" do |s|
   s.add_development_dependency 'rake'
   s.add_development_dependency 'rspec'
   s.add_development_dependency 'rubocop'
+  s.add_development_dependency 'pry-byebug'
```
Comment (gregorym) [inline bump.gemspec]: Is this necessary?

**Comment by ysv** [comment]
@grosser Actually we use bump in another project as cli and probably we will need this feature too

**Comment by grosser** [comment]
`bump next` sounds like it's modifying ... maybe `bump show-next patch` ?

File: lib/bump.rb (hunk 1 above)
Comment (ysv) [reply lib/bump.rb]: @gregorym I've moved this method to public so we can call it from bin/bump

**Comment by ysv** [comment]
@grosser Agree I've refactored to show-next

**Comment by ysv** [comment]
@grosser @gregorym Please review

File: README.md [hunk 3]
```diff
@@ -81,6 +87,7 @@ require "bump/tasks"
 ```
 
     rake bump:current                           # display current version
+    rake bump:show-next INCREMENT=minor         # dispaly next minor version
```
Comment (grosser) [inline README.md]:
```suggestion
    rake bump:show-next INCREMENT=minor         # display next minor version
```

File: lib/bump.rb [hunk 4]
```diff
@@ -39,11 +40,18 @@ def run(bump, options = {})
           bump_set(options[:version], options)
         when "current"
           ["Current version: #{current}", 0]
+        when "show-next"
+          raise InvalidIncrementError unless BUMPS.include?(options[:increment])
+
+          increment = options[:increment]
+          ["Next #{increment} version: #{next_version(current, increment)}", 0]
```
Comment (grosser) [inline lib/bump.rb]: cleaner to just return `next_version`, so result can easily be used from other scripts

File: lib/bump.rb [hunk 5]
```diff
@@ -60,6 +68,31 @@ def current
         current_info.first
       end
 
+      def next_version(current, part)
```
Comment (grosser) [inline lib/bump.rb]: part -> increment

File: lib/bump.rb [hunk 6]
```diff
@@ -60,6 +68,31 @@ def current
...
+          minor = minor.succ
+          patch = 0
+          prerelease = nil
+        when "patch"
+          patch = patch.succ
+        when "pre"
+          prerelease.strip! if prerelease.respond_to? :strip
+          prerelease = PRERELEASE[PRERELEASE.index(prerelease).succ % PRERELEASE.length]
+        else
+          raise "unknown part #{part.inspect}"
```
Comment (grosser) [inline lib/bump.rb]: raise InvalidIncrementError

**Comment by ysv** [comment]
@grosser Updated

File: README.md [hunk 7]
```diff
@@ -81,6 +87,7 @@ require "bump/tasks"
 ```
 
     rake bump:current                           # display current version
+    rake bump:show-next INCREMENT=minor         # display next minor version
```
Comment (grosser) [inline README.md]: traditionally rake tasks use underscores, prefer to stick with that even if it's not consistent

Comment (ysv) [reply README.md]: yeah yeah I know but I preferred to follow the same name as we have for bump script

**Pull request merged**